collect_app/src/test/java/org/odk/collect/android/formentry/audit/FormSaveViewModelTest.java

```diff
@@ -390,25 +394,50 @@ public void resumeFormEntry_clearsSaveResult() {
     }
 
     @Test
-    public void markOriginalFileOrDelete_whenQuestionIndexHasAnswer_onRecreatingViewModel_deletesFile() {
-        viewModel.markOriginalFileOrDelete("index", "blah");
+    public void deleteAnswerFile_whenFileHasAlreadyBeenDeleted_actuallyDeletesNewFile() {
```

Comment:
These tests about deleting/replacing media files are good to have. This behavior is hard to reason about and I don't see any context for what it's for anywhere. Maybe you could put these tests in a region and just say something like "we want to be able to go back to the exact state we were in at the last save if a user exits an instance without saving. This means keeping track of any media files that were previously attached so they can be restored." Alternately something like that could go above the two maps in `FormSaveViewModel` used for that purpose.

Also this save and discard changes behavior would be good for QA to check once the audio rework is complete. Want to keep a list like you did for the match exactly feature?

Reply:
I'll add regions/comments to try and make this clearer. Yeah I think it's a good idea to have a list of things to check!